ee/secureenclaverunner/secureenclaverunner.go

```diff
@@ -0,0 +1,282 @@
+//go:build darwin
+// +build darwin
+
+package secureenclaverunner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os/user"
+	"sync"
+	"time"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/consoleuser"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+const (
+	publicEccDataKey = "publicEccData"
+)
+
+type secureEnclaveRunner struct {
+	uidPubKeyMap        map[string]*ecdsa.PublicKey
+	secureEnclaveClient secureEnclaveClient
+	store               types.GetterSetterDeleter
+	slogger             *slog.Logger
+	mux                 *sync.Mutex
+	interrupt           chan struct{}
+	interrupted         bool
+}
+
+type secureEnclaveClient interface {
+	CreateSecureEnclaveKey(uid string) (*ecdsa.PublicKey, error)
+}
+
+func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDeleter, secureEnclaveClient secureEnclaveClient) (*secureEnclaveRunner, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	ser := &secureEnclaveRunner{
+		uidPubKeyMap:        make(map[string]*ecdsa.PublicKey),
+		store:               store,
+		secureEnclaveClient: secureEnclaveClient,
+		slogger:             slogger.With("component", "secureenclaverunner"),
+		mux:                 &sync.Mutex{},
+		interrupt:           make(chan struct{}),
+	}
+
+	data, err := store.Get([]byte(publicEccDataKey))
+	if err != nil {
+		traces.SetError(span, fmt.Errorf("getting public ecc data from store: %w", err))
+		return nil, fmt.Errorf("getting public ecc data from store: %w", err)
+	}
+
+	if data != nil {
+		if err := json.Unmarshal(data, ser); err != nil {
+			traces.SetError(span, fmt.Errorf("unmarshaling secure enclave signer: %w", err))
+			ser.slogger.Log(ctx, slog.LevelError,
+				"unable to unmarshal secure enclave signer, data may be corrupt, wiping",
+				"err", err,
+			)
+
+			if err := store.Delete([]byte(publicEccDataKey)); err != nil {
+				traces.SetError(span, fmt.Errorf("deleting corrupt public ecc data: %w", err))
+				ser.slogger.Log(ctx, slog.LevelError,
+					"unable to unmarshal secure enclave signer, data may be corrupt, wiping",
+					"err", err,
+				)
+			}
+		}
+	}
+
+	return ser, nil
+}
+
+// Public returns the public key of the current console user
+// creating and peristing a new one if needed
+func (ser *secureEnclaveRunner) Execute() error {
+	currentRetryInterval, maxRetryInterval := 1*time.Second, 1*time.Minute
+	retryTicker := time.NewTicker(currentRetryInterval)
+	defer retryTicker.Stop()
+
+	for {
+		if _, err := ser.currentConsoleUserKey(context.TODO()); err != nil {
```

Comment:
If we don't currently have a logged-in user, `currentConsoleUserKey` will return nil -- so the ticker will stop, and we will never retry creating the key until launcher itself is restarted. Is that okay? It seems like in that case we might want to keep retrying, or maybe implement something where we only start retrying once we detect a logged-in user exists?

Reply:
good catch! I'll fix that

Reply:
It looks like this is still unresolved, or am I missing the fix?